main.py

```diff
@@ -0,0 +1,97 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+@app.get("/")
+async def root():
+    return {"greeting": "Hello world"}
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorCreate,
+    db: Session = Depends(get_db),
+):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_authors(
+    skip: int = 0,
+    limit: int = 100,
+    db: Session = Depends(get_db),
+):
+    authors = crud.get_authors(db=db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Author)
+def read_author(
+    author_id: int,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author(db=db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
+
+
+# Books endpoints
```

Comment:
don't use comments like that, it doesn't give any additional info. It's just some statement.